### PR TITLE
Allow group members to view each other's contact details

### LINKED ISSUE #14: Allow group members to view each other's contact details

### PR DESCRIPTION
**Pull request by Yotam7**
closes #14 
image

### CODE REVIEW

File: codeguru/views.py [hunk 1]
```diff
@@ -75,7 +81,7 @@ def group(request, id=None):
             "is_expired": link_expired,
             "group": current_group,
             "members": members,
-            "is_in_group": (request.user.profile in members) if request.user.is_authenticated else False,
+            "is_in_group": (request.user.profile in member_profiles) if request.user.is_authenticated else False,
```
Comment (rootatkali) [inline codeguru/views.py:84]: We can just check if `user.profile.group` is the current group

File: codeguru/views.py [hunk 2]
```diff
@@ -60,7 +60,13 @@ def group(request, id=None):
     if not current_group:
         return error(request, "Group not found.")
 
-    members = Profile.objects.all().filter(group=current_group)
+    member_profiles = Profile.objects.all().filter(group=current_group)
+    members = [user for user in User.objects.all() if (user.profile in member_profiles)]
```
Comment (rootatkali) [inline codeguru/views.py:64]: There is probably a way to get the user from the profile without filtering for all users. 

Maybe [this page](https://docs.djangoproject.com/en/5.1/topics/db/examples/one_to_one/) will help

File: codeguru/views.py [hunk 3]
```diff
@@ -60,7 +60,13 @@ def group(request, id=None):
...
         return error(request, "Group not found.")
 
-    members = Profile.objects.all().filter(group=current_group)
+    member_profiles = Profile.objects.all().filter(group=current_group)
+    members = [user for user in User.objects.all() if (user.profile in member_profiles)]
+    members.sort(key=lambda member: member.username)
+
+    # Sort for group owner to be appear first. When sorting python puts False before True
+    group_owner = members[0].profile.group.owner
+    members.sort(key=lambda member: member != group_owner)
```
Comment (rootatkali) [inline codeguru/views.py:69]: Nice!

**Review by rootatkali — CHANGES_REQUESTED** [review]
Nice job! The review comments aren't big, please go over them

**Comment by rootatkali** [comment]
In addition, make sure the hebrew version works with this

File: codeguru/views.py (hunk 3 above)
Comment (Yotam7) [reply codeguru/views.py:68]: thanks

**Review by rootatkali — APPROVED** [review]
I assume you tested the changes to the group query, and also logged in as a user that isn't in the group and verified the group's details aren't visible.

If so, approved. Great work!

**Pull request merged**